### PR TITLE
Cope with missing pick-info

### PR DESCRIPTION
**Pull request by calum-chamberlain**
### What does this PR do?

Cope with missing pick.evaluation_mode as highlighted on the message-list. This is a bug in the current version of Obspy, and lots of nordic-type things have gone into master, so this ends up needing to go there.

### Why was it initiated?  Any relevant Issues?

Not all catalogues have `pick.evaluation_mode`, and shouldn't have to. The example below errors in the current version:
```python
from obspy.clients.fdsn import Client 
 
client=Client("ISC") 
cat = client.get_events(eventid=867982, includearrivals=True)  
cat.write("select.out", format="NORDIC")
```

This PR fixes this, and patches an issue with missing weights (with long-phase names) and polarities.  It also changes mapping of an empty polarity field from `undecidable` to `None` when reading nordic files.

### PR Checklist
- [ ] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [x] This PR is not directly related to an existing issue (which has no PR yet).
~~- [ ] If the PR is making changes to documentation, docs pages can be built automatically.~~
~~      Just remove the space in the following string after the + sign: "+ DOCS"~~
- [ ] If any network modules should be tested for the PR, add them as a comma separated list
      (e.g. `clients.fdsn,clients.arclink`) after the colon in the following magic string: "+TESTS:"
      (you can also add "ALL" to just simply run all tests across all modules)
- [x] All tests still pass.
- [x] Any new features or fixed regressions are be covered via new tests.
~~- [ ] Any new or changed features have are fully documented.~~
- [x] Significant changes have been added to `CHANGELOG.txt` .
~~- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .~~

### CODE REVIEW

**Comment by megies** [comment]
- I was a bit wary about the default `verbose=True` in there, but it turns out sth is printed only if an exception is raised afterwards. In such a case I would just pass the error message as the `return` value as opposed to `True` if no error, but thats just cosmetics, just fyi
 - CI fails unrelated
 -  [ ] changelog: IIRC this is a bug only in master? please get rid of changelog message if thats correct
 - [ ] warnings need to be caught in tests:

```
........................../Users/travis/miniconda/envs/test-environment/lib/python3.7/site-packages/obspy/io/nordic/core.py:1450: UserWarning: Evaluation mode None is not mappable

  pick.evaluation_mode))

...../Users/travis/miniconda/envs/test-environment/lib/python3.7/site-packages/obspy/io/nordic/core.py:1450: UserWarning: Evaluation mode None is not mappable

  pick.evaluation_mode))

...../Users/travis/miniconda/envs/test-environment/lib/python3.7/site-packages/obspy/io/nordic/core.py:1450: UserWarning: Evaluation mode None is not mappable

  pick.evaluation_mode))

............/Users/travis/miniconda/envs/test-environment/lib/python3.7/site-packages/obspy/io/nordic/core.py:1450: UserWarning: Evaluation mode None is not mappable

  pick.evaluation_mode))

../Users/travis/miniconda/envs/test-environment/lib/python3.7/site-packages/obspy/io/nordic/core.py:1450: UserWarning: Evaluation mode None is not mappable

  pick.evaluation_mode))

......../Users/travis/miniconda/envs/test-environment/lib/python3.7/site-packages/obspy/io/nordic/core.py:1450: UserWarning: Evaluation mode None is not mappable

  pick.evaluation_mode))
```

**Comment by calum-chamberlain** [comment]
Thanks @megies will make those changes now - leaving `verbose=True` was an oversight (again!).  This bug is also in the 1.1.1 release, so I will leave the note in the changelog.  A user on the email list ran into it using their installed version.

**Comment by calum-chamberlain** [comment]
The comparison test now returns a useful string which is raised by the assert similarity checker, warnings are caught (adding in the less complete pick added a lot of warnings, which I think are justified because you are effectively losing information by converting to Nordic), and changelog entry remains - maybe good to go.

Sorry for adding more to the 1.2.0 list!

**Comment by ThomasLecocq** [comment]
@calum-chamberlain
- the base branch is: **master**
- there is no need for network tests in your case

for the rest the CI looks happy

File: CHANGELOG.txt [hunk 1]
```diff
@@ -117,6 +117,7 @@ master:
    * Include AIN as takeoff-angle when reading and writing nordic files
      (see #2404).
    * Add error ellipses and read high-accuracy hypocenter lines (see #2451)
+   * BUG-FIX: cope with missing pick evaluation information (see #2520)
```
Comment (ThomasLecocq) [inline CHANGELOG.txt]: could maybe be a little more clear here, like

```
* Fix the incorrect handling of events missing pick evaluation information (see #2520)
```

**Pull request merged**